pygmt/src/_common.py

```diff
@@ -251,3 +251,56 @@ def from_params(
             f"{', '.join(params)}."
         )
         raise GMTInvalidInput(msg)
+
+
+def _parse_coastline_resolution(
+    resolution: Literal["auto", "full", "high", "intermediate", "low", "crude", None],
+) -> Literal["a", "f", "h", "i", "l", "c", None]:
```

Comment:
Thoughts on making a StrEnum out of this?

Reply:
What are the benefits? This private function will likely be removed after #3239 is implemented.

Reply:
GMT uses short-form arguments, but we prefer long-form arguments in PyGMT. Another example is in #3012. It will be a lot of work if we define StrEnum for all these cases. #3239 proposed a more general solution for it.

Reply:
Yeah, `StrEnum` will be a lot more work, and we'll need to bring in something like pydantic (mentioned at https://github.com/GenericMappingTools/pygmt/issues/3239#issuecomment-2126307579) to fully enable runtime checks. Ok to keep this as is for now, and wait for #3239.